packages/create-react-native-library/templates/common/$package.json

```diff
@@ -72,7 +78,7 @@
   "devDependencies": {
     "@commitlint/config-conventional": "^17.0.2",
     "@evilmartians/lefthook": "^1.5.0",
-<% if (example === 'vanilla') { -%>
+<% if (example === 'vanilla' && !project.nitro) { -%>
```

Comment:
It'll be more future-proof to add configs based on what type of project it is rather than what type of project it is not. That way adding a new type won't require adding new condition to each call site.

I think we should just add a key for the official module type and use it for these checks (here and other places) instead of `!project.nitro`.

Reply:
I've introduced `project.moduleConfig` (which replaces `project.module`) as an union type of `native-modules`, `turbo-modules`, `nitro-modules`. This is the native modules api that we use.